### DISCUSSION TITLE
Mark collection as 'don't destroy on load'

### DISCUSSION THREAD

**Discussion by alexanderameye** [neon-age/Smart-Hierarchy#12] [Ideas]
Right now, when I have a 'don't destroy on load' object under a collection, I get the following warning from Unity.

`DontDestroyOnLoad only works for root GameObjects or components on root GameObjects.
UnityEngine.Object:DontDestroyOnLoad(Object)`

Would it be possible to allow us the option to mark a collection as being 'persistent' so that everything under that collection is marked as DDOL?

**Edit:** Right after typing this, I did see that the collection object is just a gameobject that I can add components to, so I could as well put a DDOL script on it! If no further discussion is deemed necessary, this can be closed :)

**Comment by neon-age** [comment]
Collection object is being destroyed before entering playmode or during build process (see [https://github.com/neon-age/Smart-Hierarchy/blob/master/Collections/DetachOnBuildProcess.cs](DetachOnBuildProcess.cs). This mean that any attached component will be stripped from build.

Should I add an option like "Keep Transform Hierarchy" for the cases where transform overhead is not true (eg. collection of static objects/managers)?
It would look something like this:
![image](https://user-images.githubusercontent.com/29812914/106749732-7bcbac00-6659-11eb-8fd1-2586bd871ee1.png)

Another thing that comes to my mind is "Add Components To Root Children" which will copy/paste components to root children during scene processing. Though it's harder to implement nicely, and I'm not sure if that's a good idea..

**Reply by alexanderameye** [reply]
Yeah static managers is exactly my use case and something like that toggle would be nice indeed so it can be persistent. The overhead of using the collection and keeping it, would be the same as if I were to create an empty gameobject as a parent? Or does the collection parent have any additional overhead?

**Reply by neon-age** [reply]
Yes, it's the same as using GameObject as a parent. 
I'll push this feature right now, let me know if you have any other feedback!
